src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -63346,6 +63346,51 @@ void f4(object? x4)
                 Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "x4").WithLocation(36, 13));
         }
 
+        [Fact, WorkItem(65061, "https://github.com/dotnet/roslyn/issues/65061")]
+        public void LocalFunctions_Speculation()
+        {
+            var source1 = """
+                #nullable enable
+                void func() { object? o = null; }
+                void target() { object? o = null; }
+                """;
+            var source2 = """
+                void target() { object? o = ""; }
+                """;
+
+            var comp1 = CreateCompilation(source1).VerifyDiagnostics(
+                // (2,6): warning CS8321: The local function 'func' is declared but never used
```

Comment:
>// (2,6): warning CS8321: The local function 'func' is declared but never used [](http://example.com/codeflow?start=16&length=78)

Could we suppress warnings with pragma to reduce the noise? Or they are needed for the repro?